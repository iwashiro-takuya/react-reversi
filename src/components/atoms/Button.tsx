import React from 'react';
import styled from 'styled-components';
import Const from 'src/const';

const { Color } = Const;

export interface ButtonProp {
  onClick: () => void;
  dataCy: string;
}
const Button: React.FC<ButtonProp> = ({ children, onClick, dataCy }) => {
  return (
    <StyledBottun type="button" onClick={onClick} data-cy={dataCy}>
      {children}
    </StyledBottun>
  );
};
export default Button;

const StyledBottun = styled.button`
  display: block;
  padding: 8px 16px;
  margin: 0 auto;
  background-color: ${Color.BG_WHITE};
  border: 1px solid ${Color.BD_BLACK};
  border-radius: 4px;
`;
