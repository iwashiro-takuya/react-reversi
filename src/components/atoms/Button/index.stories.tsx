import React from 'react';
import styled from 'styled-components';
import { Story } from '@storybook/react/types-6-0';
import Button, { ButtonProp } from './component';

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    // creates a specific argType based on the iconMap object
    text: {
      control: {
        type: 'text',
      },
    },
  },
};

const Template: Story<ButtonProp> = (args) => (
  <Base>
    <Button {...args}>{args.text}</Button>
  </Base>
);

export const normal = Template.bind({});

normal.args = {
  text: 'Game Start',
};

const Base = styled.div`
  width: fit-content;
`;
