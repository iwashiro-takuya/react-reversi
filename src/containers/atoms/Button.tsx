import React from 'react';
import Button from 'src/components/atoms/Button';

interface EnhancedButtonProps {
  onClick: () => void;
  text: string;
}
const EnhancedButton: React.FC<EnhancedButtonProps> = ({ onClick, text }) => {
  return <Button onClick={onClick}>{text}</Button>;
};
export default EnhancedButton;
