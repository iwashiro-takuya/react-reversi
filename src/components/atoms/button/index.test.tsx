import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';

describe('Button unit test', () => {
  it('snapshot', () => {
    const component = renderer.create(
      <Button
        onClick={() => {
          return true;
        }}
        text="Game Start"
        dataCy="start"
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
