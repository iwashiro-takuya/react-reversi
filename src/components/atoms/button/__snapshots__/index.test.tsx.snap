// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button unit test snapshot 1`] = `
<button
  className="sc-bdnxRM jTgbIw"
  data-cy="start"
  onClick={[Function]}
  type="button"
>
  Game Start
</button>
`;
